HardwareIntrinsics test build crashes intermittently

## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=1154397&view=logs&j=2c125651-5e8d-5bee-571d-64eeebde9b33
Build error leg or test failing: Build / linux-arm checked CoreCLR_ReleaseLibraries / Build native test components
Pull request: https://github.com/dotnet/runtime/pull/119937
<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "",
  "ErrorPattern": "HardwareIntrinsics.*exited with code 1",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

This is hit with other projects as well. From https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/1154732/logs/832 :
```
2025-09-22T09:46:04.7628392Z /__w/1/s/src/tests/build.proj(426,5): error MSB3073: The command ""/__w/1/s/.dotnet/dotnet" restore  -r linux-x64 Common/XHarnessRunnerLibrary/XHarnessRunnerLibrary.csproj  /p:SetTFMForRestore=true /p:RuntimeIdentifier=linux-x64 /p:TargetOS=linux /p:TargetArchitecture=x64 /p:Configuration=Checked /p:CrossBuild=true" exited with code 1.
```

--- a/eng/build.ps1
+++ b/eng/build.ps1
@@ -266,16 +266,6 @@ if ($vs) {
   # Disable .NET runtime signature validation errors which errors for local builds
   $env:VSDebugger_ValidateDotnetDebugLibSignatures=0;
 
-  # MSBuild Server dogfood.
-  if ($env:DOTNET_CLI_USE_MSBUILD_SERVER -eq $null) 
-  {
-    $env:DOTNET_CLI_USE_MSBUILD_SERVER = 1
-  }
-  if ($env:MSBUILDUSESERVER -eq $null) 
-  {
-    $env:MSBUILDUSESERVER = 1
-  }
-
   # Respect the RuntimeConfiguration variable for building inside VS with different runtime configurations
   if ($runtimeConfiguration)
   {

--- a/eng/build.sh
+++ b/eng/build.sh
@@ -14,10 +14,6 @@ while [[ -h "$source" ]]; do
 done
 scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
 
-# MSBuild Server dogfood.
-export MSBUILDUSESERVER="${MSBUILDUSESERVER:-1}"
-export DOTNET_CLI_USE_MSBUILD_SERVER="${DOTNET_CLI_USE_MSBUILD_SERVER:-1}"
-
 usage()
 {
   echo "Common settings:"

--- a/eng/pipelines/common/global-build-job.yml
+++ b/eng/pipelines/common/global-build-job.yml
@@ -81,10 +81,6 @@ jobs:
         - ${{ format('build_{0}{1}_{2}_{3}_{4}', parameters.osGroup, parameters.osSubgroup, parameters.archType, coalesce(globalBuild.buildConfig, parameters.buildConfig), globalBuild.nameSuffix) }}
 
     variables:
-      - name: "MSBUILDUSESERVER"
-        value: "1"
-      - name: "DOTNET_CLI_USE_MSBUILD_SERVER"
-        value: "1"
       - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
           - group: DotNet-HelixApi-Access
           - group: AzureDevOps-Artifact-Feeds-Pats

--- a/eng/pipelines/common/templates/global-build-step.yml
+++ b/eng/pipelines/common/templates/global-build-step.yml
@@ -12,9 +12,6 @@ parameters:
 steps:
   - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -ci ${{ parameters.archParameter }} $(_osParameter) ${{ parameters.crossArg }} ${{ parameters.buildArgs }} ${{ parameters.targetCxxLibraryConfigurationArgs }} $(_officialBuildParameter) $(_overrideTestScriptWindowsCmdParameter)
     displayName: ${{ parameters.displayName }}
-    env:
-      MSBUILDUSESERVER: '1'
-      DOTNET_CLI_USE_MSBUILD_SERVER: '1'
     ${{ if eq(parameters.useContinueOnErrorDuringBuild, true) }}:
       continueOnError: ${{ parameters.shouldContinueOnError }}
     ${{ if ne(parameters.container, '') }}:

--- a/eng/pipelines/global-build.yml
+++ b/eng/pipelines/global-build.yml
@@ -60,9 +60,6 @@ extends:
                 eq(stageDependencies.EvaluatePaths.evaluate_paths.outputs['SetPathVars_coreclr.containsChange'], true),
                 eq(stageDependencies.EvaluatePaths.evaluate_paths.outputs['SetPathVars_non_mono_and_wasm.containsChange'], true),
                 eq(variables['isRollingBuild'], true))
-          env:
-            MSBUILDUSESERVER: 1
-            DOTNET_CLI_USE_MSBUILD_SERVER: 1
 
       #
       # Build with RuntimeFlavor only. This exercise code paths where only RuntimeFlavor is

